src/vmm/src/persist.rs

```diff
@@ -203,14 +206,20 @@ pub fn create_snapshot(
         .save_state()
         .map_err(CreateSnapshotError::MicrovmState)?;
 
+    #[cfg(target_arch = "x86_64")]
+    let snapshot_data_version =
+        get_snapshot_data_version(&params.version, &version_map, &vmm.mmio_device_manager)?;
+
+    #[cfg(target_arch = "aarch64")]
+    let snapshot_data_version = get_snapshot_data_version(&params.version, &version_map)?;
+
```

Comment:
```suggestion
    let snapshot_data_version =
        get_snapshot_data_version(&params.version, &version_map, &vmm)?;
```
when switching to using `vmm` as param.